src/Estimators/include/BipedalLocomotion/Estimators/FloatingBaseEstimator.h

```diff
@@ -0,0 +1,268 @@
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler, const iDynTree::Model& model);
+
+    /**
+    * Set the polled IMU measurement
+    * @param[in] acc_meas linear accelerometer measurement expressed in the local IMU frame (m/s^2)
+    * @param[in] gyro_meas gyroscope measurement expressed in the local IMU frame (rad/s)
+    * @return True in case of success, false otherwise.
+    */
+    bool setIMUMeasurement(const Eigen::Vector3d& acc_meas,
+                           const Eigen::Vector3d& gyro_meas);
+
+    /**
+    * Set feet contact states
+    * @param[in] lf_in_contact left foot contact state [0, 1]
+    * @param[in] rf_in_contact right foot contact state [0, 1]
+    * @return True in case of success, false otherwise.
+    */
+    bool setContacts(const bool& lf_in_contact, const bool& rf_in_contact);
+
+    /**
+    * Set kinematic measurements
+    * @note it is assumed that the order of the joints loaded in the model and the order of the measurements in these vectors match
+    * @param[in] encoders joint positions measured through encoders
+    * @param[in] encoder_speeds joint velocities measured through encoders
+    * @return True in case of success, false otherwise.
+    */
+    bool setKinematics(const Eigen::VectorXd& encoders,
+                       const Eigen::VectorXd& encoder_speeds);
+
+    /**
+    * Compute one step of the estimator
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool advance() final;
+
+    /**
+    * Get estimator outputs
+    * @return A struct containing he estimated internal states of the estiamtor and the associated covariance matrix
+    */
+    virtual const FloatingBaseEstimators::Output& get() const final;
+
+    /**
+    * Determines the validity of the object retrieved with get()
+    * @return True in case of success, false otherwise.
+    */
+    virtual bool isValid() const final { return (m_estimator_state == State::Running); };
+
+    /**
+    * Get ModelComputations object by reference
+    * @return ModelComputations object providing information between considered model related quantities in the estimator
+    * like the base link, IMU, feet contact frames.
+    */
+    ModelComputations& modelComputations();
+
+protected:
+    /**
+    * Configure generic parameters
+    * @param[in] handler configure the generic parameters for the estimator
+    * @return True in case of success, false otherwise.
+    */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
+
+    /**
+    * These custom parameter specifications should be specified by the derived class.
+    * @param[in] handler configure the custom parameters for the estimator
+    * @return bool
+    */
+    virtual bool customInitialization(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler) { return true; };
+
```

Comment:
already implement and add default setup() methods for configuring `PriorStdDev`, `Options`, `InitialState` and `SensorsStdDev`  as protected methods. This way these methods can be chosen to be used or not used within the `customInitialization()`. This keeps most of the required configuration within the base class and the derived class can mainly focus on algorithmic implementation.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/79/commits/f46bd5b271b6e7ea77da83c41b90178157aac069